File: src/Entity/ShippingAddresses.php
<?php

namespace App\Entity;

use Doctrine\ORM\Mapping as ORM;

/**
 * ShippingAddresses
 *
 * @ORM\Table(name="shipping_addresses", indexes={@ORM\Index(name="shp_user_id", columns={"shp_user_id"}), @ORM\Index(name="shp_country_id", columns={"shp_country_id"})})
 * @ORM\Entity
 */
class ShippingAddresses
{
    /**
     * @var int
     *
     * @ORM\Column(name="shp_id", type="bigint", nullable=false, options={"unsigned"=true})
     * @ORM\Id
     * @ORM\GeneratedValue(strategy="IDENTITY")
     */
    private $shpId;

    /**
     * @var string
     *
     * @ORM\Column(name="shp_city", type="string", length=55, nullable=false)
     */
    private $shpCity;

    /**
     * @var string
     *
     * @ORM\Column(name="shp_region", type="string", length=55, nullable=false)
     */
    private $shpRegion;

    /**
     * @var string
     *
     * @ORM\Column(name="shp_street", type="string", length=55, nullable=false)
     */
    private $shpStreet;

    /**
     * @var string
     *
     * @ORM\Column(name="shp_street_no", type="string", length=8, nullable=false)
     */
    private $shpStreetNo;

    /**
     * @var string
     *
     * @ORM\Column(name="shp_code", type="string", length=9, nullable=false)
     */
    private $shpCode;

    /**
     * @var string
     *
     * @ORM\Column(name="shp_surname", type="string", length=55, nullable=false)
     */
    private $shpSurname;

    /**
     * @var string
     *
     * @ORM\Column(name="shp_name", type="string", length=55, nullable=false)
     */
    private $shpName;

    /**
     * @var \Users
     *
     * @ORM\ManyToOne(targetEntity="Users")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="shp_user_id", referencedColumnName="id")
     * })
     */
    private $shpUser;

    /**
     * @var \Countries
     *
     * @ORM\ManyToOne(targetEntity="Countries")
     * @ORM\JoinColumns({
     *   @ORM\JoinColumn(name="shp_country_id", referencedColumnName="ctr_id")
     * })
     */
    private $shpCountry;

    public function getShpId(): ?string
    {
        return $this->shpId;
    }

    public function getShpCity(): ?string
    {
        return $this->shpCity;
    }

    public function setShpCity(string $shpCity): self
    {
        $this->shpCity = $shpCity;

        return $this;
    }

    public function getShpRegion(): ?string
    {
        return $this->shpRegion;
    }

    public function setShpRegion(string $shpRegion): self
    {
        $this->shpRegion = $shpRegion;

        return $this;
    }

    public function getShpStreet(): ?string
    {
        return $this->shpStreet;
    }

    public function setShpStreet(string $shpStreet): self
    {
        $this->shpStreet = $shpStreet;

        return $this;
    }

    public function getShpStreetNo(): ?string
    {
        return $this->shpStreetNo;
    }

    public function setShpStreetNo(string $shpStreetNo): self
    {
        $this->shpStreetNo = $shpStreetNo;

        return $this;
    }

    public function getShpCode(): ?string
    {
        return $this->shpCode;
    }

    public function setShpCode(string $shpCode): self
    {
        $this->shpCode = $shpCode;

        return $this;
    }

    public function getShpSurname(): ?string
    {
        return $this->shpSurname;
    }

    public function setShpSurname(string $shpSurname): self
    {
        $this->shpSurname = $shpSurname;

        return $this;
    }

    public function getShpName(): ?string
    {
        return $this->shpName;
    }

    public function setShpName(string $shpName): self
    {
        $this->shpName = $shpName;

        return $this;
    }

    public function getShpUser(): ?Users
    {
        return $this->shpUser;
    }

    public function setShpUser(?Users $shpUser): self
    {
        $this->shpUser = $shpUser;

        return $this;
    }

    public function getShpCountry(): ?Countries
    {
        return $this->shpCountry;
    }

    public function setShpCountry(?Countries $shpCountry): self
    {
        $this->shpCountry = $shpCountry;

        return $this;
    }


}

File: src/main/java/models/users/ShippingAddresses.java
package models.users;

import lombok.Getter;
import lombok.Setter;
import models.orders.Orders;
import models.general.Countries;

import java.io.Serializable;

import javax.persistence.*;
import java.util.Collection;



import javax.xml.bind.annotation.XmlTransient;

@Entity
@Table(name = "shipping_addresses")
public class ShippingAddresses implements Serializable {

    private static final long serialVersionUID = 1L;
    @Getter
    @Setter
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column
    private Long id;

    @Getter
    @Setter
    @Column
    private String city;

    @Getter
    @Setter
    @Column
    private String region;

    @Getter
    @Setter
    @Column
    private String street;

    @Getter
    @Setter
    @Column
    private String streetNo;

    @Getter
    @Setter
    @Column
    private String code;

    @Getter
    @Setter
    @Column
    private String surname;

    @Getter
    @Setter
    @Column
    private String name;

    @JoinColumn(name = "shp_user_id", referencedColumnName = "id")
    @ManyToOne(optional = false, fetch = FetchType.LAZY)
    private Users userObj;

    @JoinColumn(name = "shp_country_id", referencedColumnName = "ctr_id")
    @ManyToOne(optional = false, fetch = FetchType.LAZY)

    private Countries countryObj;

    @OneToMany(cascade = CascadeType.ALL, mappedBy = "shipAddressObj", fetch = FetchType.LAZY)
    private Collection<Orders> ordersCollection;

    public ShippingAddresses() {
    }

    public ShippingAddresses(Long shpId) {
        this.id = shpId;
    }

    public ShippingAddresses(Long shpId, String shpCity, String shpRegion, String shpStreet, String shpStreetNo, String shpCode, String shpSurname, String shpName) {
        this.id = shpId;
        this.city = shpCity;
        this.region = shpRegion;
        this.street = shpStreet;
        this.streetNo = shpStreetNo;
        this.code = shpCode;
        this.surname = shpSurname;
        this.name = shpName;
    }

    public Users getUserObj() {
        return userObj;
    }

    public void setUserObj(Users userObj) {
        this.userObj = userObj;
    }

    public Countries getCountryObj() {
        return countryObj;
    }

    public void setCountryObj(Countries countryObj) {
        this.countryObj = countryObj;
    }

    @XmlTransient
    public Collection<Orders> getOrdersCollection() {
        return ordersCollection;
    }

    public void setOrdersCollection(Collection<Orders> ordersCollection) {
        this.ordersCollection = ordersCollection;
    }
}
